http-clients-api/src/main/java/com/palantir/remoting1/clients/ClientConfig.java

```diff
@@ -65,20 +60,12 @@ public Integer maxNumRetries() {
         return MAX_NUM_RETRIES;
     }
 
-    @Value.Check
-    public void check() {
-        Preconditions.checkState(sslSocketFactory().isPresent() == trustManager().isPresent(),
-                "Must set either both SslSocketFactory and TrustManager, or neither");
-    }
-
     public static ClientConfig fromServiceConfig(ServiceConfiguration serviceConfig) {
         ClientConfig.Builder clientConfig = builder();
 
         // ssl
         if (serviceConfig.security().isPresent()) {
-            clientConfig.sslSocketFactory(SslSocketFactories.createSslSocketFactory(serviceConfig.security().get()));
-            clientConfig.trustManager(
-                    (X509TrustManager) SslSocketFactories.createTrustManagers(serviceConfig.security().get())[0]);
+            clientConfig.trustContext(SslSocketFactories.createTrustContext(serviceConfig.security().get()));
```

Comment:
I think you're comment on this atlas pr is similar to what I'm talking about now: https://github.com/palantir/atlasdb/pull/650/files/a387f0ad029979fbfa818d9a323cb3eb479aec5b..9921ad880eec217bd90267e11cf35706e1165fd9#diff-eac7e82cc97989b4f7c7451d5b12087dR96